docs: Go fasthttp package documentation (#1132)
getsentry/sentry-docs: src/collections/_documentation/platforms/go/fasthttp.md

---
title: Fasthttp
---

# Official Sentry fasthttp Handler for Sentry-go SDK

**Godoc:** https://godoc.org/github.com/getsentry/sentry-go/fasthttp

**Example:** https://github.com/getsentry/sentry-go/tree/master/example/fasthttp

## Installation

```sh
go get github.com/getsentry/sentry-go/fasthttp
```

```go
import (
	"fmt"
	"net/http"

	"github.com/getsentry/sentry-go"
	sentryfasthttp "github.com/getsentry/sentry-go/fasthttp"
)

// To initialize Sentry's handler, you need to initialize Sentry itself beforehand
if err := sentry.Init(sentry.ClientOptions{
	Dsn: "your-public-dsn",
}); err != nil {
	fmt.Printf("Sentry initialization failed: %v\n", err)
}

// Create an instance of sentryfasthttp
sentryHandler := sentryfasthttp.New(sentryfasthttp.Options{})

> // Once it's done, you can attach the handler as one of your middlewares

Reviewer: ```suggestion
// After creating the instance, you can attach the handler as one of your middleware
```

fastHTTPHandler := sentryHandler.Handle(func(ctx *fasthttp.RequestCtx) {
	panic("y tho")
})

fmt.Println("Listening and serving HTTP on :3000")

// And run it
if err := fasthttp.ListenAndServe(":3000", fastHTTPHandler); err != nil {
	panic(err)
}
```

## Configuration

`sentryfasthttp` accepts a struct of `Options` that allows you to configure how the handler will behave.

> Currently it respects 3 options:

Reviewer: ```suggestion
Currently, it respects three options:
```

```go

> // Repanic configures whether Sentry should repanic after recovery, in most cases it should be set to false,

Reviewer: ```suggestion
// Repanic configures whether Sentry should repanic after recovery, in most cases, it defaults to false,
```

> // as fasthttp doesn't include it's own Recovery handler.

Reviewer: ```suggestion
// as fasthttp doesn't include its own Recovery handler.
```

Repanic bool
// WaitForDelivery configures whether you want to block the request before moving forward with the response.

> // Because fasthttp doesn't include it's own `Recovery` handler, it will restart the application,

Reviewer: ```suggestion
// Because fasthttp doesn't include its own `Recovery` handler, it will restart the application,
```

> // and event won't be delivered otherwise.

Reviewer: ```suggestion
// and the event won't be delivered otherwise.
```

WaitForDelivery bool
// Timeout for the event delivery requests.
Timeout time.Duration
```

## Usage

`sentryfasthttp` attaches an instance of `*sentry.Hub` (https://godoc.org/github.com/getsentry/sentry-go#Hub) to the request's context, which makes it available throughout the rest of the request's lifetime.
You can access it by using the `sentryfasthttp.GetHubFromContext()` method on the context itself in any of your proceeding middleware and routes.
And it should be used instead of the global `sentry.CaptureMessage`, `sentry.CaptureException`, or any other calls, as it keeps the separation of data between the requests.

> **Keep in mind that `*sentry.Hub` won't be available in middleware attached before to `sentryfasthttp`!**

Reviewer: ```suggestion
**Keep in mind that `*sentry.Hub` won't be available in middleware attached before `sentryfasthttp`!**
```
